hibernate-core/src/main/java/org/hibernate/dialect/function/CommonFunctionFactory.java

```diff
@@ -2379,12 +2384,33 @@ public static void format_toVarchar(QueryEngine queryEngine) {
 				.setInvariantType(
 						queryEngine.getTypeConfiguration().getBasicTypeRegistry().resolve( StandardBasicTypes.STRING )
 				)
-				.setExactArgumentCount( 2 )
-				.setParameterTypes(TEMPORAL, STRING)
+				.setArgumentsValidator( formatValidator() )
 				.setArgumentListSignature( "(TEMPORAL datetime as STRING pattern)" )
 				.register();
 	}
 
+	public static ArgumentsValidator formatValidator() {
```

Comment:
This is a good idea, but have fun with the conflicts.

Reply:
I fixed them already :)

Reply:
Oh, I see, no conflicts.